cgi-bin/DW/Hooks/SpamCheck.pm

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/perl
+#
+# DW::Hooks::SpamCheck
+#
+# This module implements a hook for checking input for blocked domains, and
+# auto-suspending a user if one is found.
+#
+# Authors:
+#      Momiji <momijizukamori@gmail.com>
+#
+# Copyright (c) 2023 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Hooks::SpamCheck;
+
+use strict;
+use warnings;
+use Scalar::Util qw/reftype/;
+
+use LJ::Hooks;
+
+LJ::Hooks::register_hook(
+    'spam_check',
+    sub {
+        my ( $u, $data, $location ) = @_;
+        return unless defined $u && defined $data && defined $location;
+        return if $u->has_priv('siteadmin');    # some users can be trusted
```

Comment:
I definitely misread this as "can't be trusted" and was like "you know, she's right."